src/ripple/app/misc/impl/ValidatorSite.cpp

```diff
@@ -38,13 +39,21 @@ ValidatorSite::Site::Resource::Resource (std::string uri_)
     : uri {std::move(uri_)}
 {
     if (! parseUrl (pUrl, uri) ||
-        (pUrl.scheme != "http" && pUrl.scheme != "https"))
+        (pUrl.scheme != "http" && pUrl.scheme != "https"
```

Comment:
My personal preference would be to have this be. I think it reads easier. YMMV.

```
if (! parseUrl (pUrl, uri))
    throw std::runtime_error("URI '" + uri + "' cannot be parsed");

if  (pUrl.scheme == "file")
{
    if (!pUrl.domain.empty())
        throw std::runtime_error("file URI cannot contain a hostname");
}
else if (pUrl.scheme == "http")
{
    if (pUrl.domain.empty())
        throw std::runtime_error("http URI must contain a hostname");

    if (!pUrl.port)
        pUrl.port = 80;
}
else if (pUrl.scheme == "https")
{
    if (pUrl.domain.empty())
        throw std::runtime_error("https URI must contain a hostname");

    if (!pUrl.port)
        pUrl.port = 443;
}
else
    throw std::runtime_error ("Unsupported scheme: '" + pUrl.scheme + "'");
```

Reply:
Yeah, I like that. Much more readable.